app/code/Magento/Catalog/Model/ImageUploader.php

```diff
@@ -276,4 +280,22 @@ public function saveFileToTmpDir($fileId)
 
         return $result;
     }
+
+    /**
+     * Get the final filename to use when the image is saved
+     * @param string $imageName
+     * @return string
+     */
+    public function getFinalImageName($imageName)
```

Comment:
It is the backward incompatible breaking change (which also is impossible to backport to Magento 2.2). Consider refactoring the code to eliminate it. Please refer to the guide https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/

Reply:
Sorry, I missed that this class is not marked with`@api`. So you can introduce a public method here